[loguru] Add feature to link with fmt

**Is your feature request related to a problem? Please describe.**
Loguru supports string formatting with fmt via `#define LOGURU_USE_FMTLIB 1`. However, when trying to use this feature with non-header-only, it fails with linker error
``undefined reference to `loguru::vlog(int, char const*, unsigned int, char const*, fmt::v8::basic_format_args<fmt::v8::basic_format_context<fmt::v8::appender, char> >)'``, which I am guessing comes from the loguru binary not being linked to the fmt library when it is built.

**Proposed solution**
I suggest adding a feature to loguru that has a dependency on fmt and that will build loguru linked to fmt. I am already working on a solution on my [fork](https://github.com/mrlegohead0x45/vcpkg) and I will create a pr if people agree.

**Describe alternatives you've considered**
Use loguru with header only.


Also slightly off topic, but why is non-header-only not supported on Windows?

--- a/ports/loguru/CMakeLists.txt
+++ b/ports/loguru/CMakeLists.txt
@@ -19,7 +19,7 @@ target_include_directories(loguru PUBLIC
                                $<INSTALL_INTERFACE:include/loguru>)
 
 if(BUILD_WITH_FMT)
-    target_compile_definitions(loguru PRIVATE LOGURU_USE_FMTLIB)
+    target_compile_definitions(loguru PRIVATE LOGURU_USE_FMTLIB=1)
     find_package(fmt REQUIRED)
     target_link_libraries(loguru PUBLIC fmt::fmt)
 endif()

--- a/ports/loguru/CMakeLists.txt
+++ b/ports/loguru/CMakeLists.txt
@@ -4,6 +4,7 @@ project(loguru CXX)
 set(CMAKE_CXX_STANDARD 11)
 
 option(INSTALL_HEADERS "Install header files" ON)
+option(BUILD_WITH_FMT  "Build with fmt support" OFF)
 
 set(loguru_headers
   loguru.hpp
@@ -16,6 +17,12 @@ add_library(loguru ${loguru_sources})
 target_include_directories(loguru PUBLIC 
                                $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
                                $<INSTALL_INTERFACE:include/loguru>)
+
+if(BUILD_WITH_FMT)
+    target_compile_definitions(loguru PRIVATE LOGURU_USE_FMTLIB)
+    find_package(fmt REQUIRED)
+    target_link_libraries(loguru PUBLIC fmt::fmt)
+endif()
 
 if(UNIX)
     set(THREADS_PREFER_PTHREAD_FLAG ON)

--- a/ports/loguru/CMakeLists.txt
+++ b/ports/loguru/CMakeLists.txt
@@ -20,7 +20,7 @@ target_include_directories(loguru PUBLIC
 
 if(BUILD_WITH_FMT)
     target_compile_definitions(loguru PRIVATE LOGURU_USE_FMTLIB=1)
-    find_package(fmt REQUIRED)
+    find_package(fmt CONFIG REQUIRED)
     target_link_libraries(loguru PUBLIC fmt::fmt)
 endif()
 

--- a/ports/loguru/portfile.cmake
+++ b/ports/loguru/portfile.cmake
@@ -21,9 +21,8 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
             fmt BUILD_WITH_FMT
     )
 
-    vcpkg_configure_cmake(
+    vcpkg_cmake_configure(
         SOURCE_PATH ${SOURCE_PATH}
-        PREFER_NINJA
         OPTIONS_DEBUG
             -DINSTALL_HEADERS=OFF
 
@@ -31,8 +30,8 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
             ${FEATURE_OPTIONS}
      )
 
-    vcpkg_install_cmake()
-    vcpkg_fixup_cmake_targets()
+    vcpkg_cmake_install()
+    vcpkg_cmake_config_fixup()
     vcpkg_copy_pdbs()
 endif()
 

--- a/ports/loguru/portfile.cmake
+++ b/ports/loguru/portfile.cmake
@@ -22,10 +22,9 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     )
 
     vcpkg_cmake_configure(
-        SOURCE_PATH ${SOURCE_PATH}
+        SOURCE_PATH "${SOURCE_PATH}"
         OPTIONS_DEBUG
             -DINSTALL_HEADERS=OFF
-
         OPTIONS
             ${FEATURE_OPTIONS}
      )

--- a/ports/loguru/portfile.cmake
+++ b/ports/loguru/portfile.cmake
@@ -15,11 +15,20 @@ if(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
     configure_file(${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt ${SOURCE_PATH}/CMakeLists.txt COPYONLY)
     configure_file(${CMAKE_CURRENT_LIST_DIR}/loguruConfig.cmake.in  ${SOURCE_PATH}/loguruConfig.cmake.in COPYONLY)
 
+    vcpkg_check_features(
+        OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+        FEATURES
+            fmt BUILD_WITH_FMT
+    )
+
     vcpkg_configure_cmake(
         SOURCE_PATH ${SOURCE_PATH}
         PREFER_NINJA
         OPTIONS_DEBUG
             -DINSTALL_HEADERS=OFF
+
+        OPTIONS
+            ${FEATURE_OPTIONS}
      )
 
     vcpkg_install_cmake()

--- a/ports/loguru/vcpkg.json
+++ b/ports/loguru/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "loguru",
-  "version-string": "2.1.0",
+  "version": "2.1.0",
   "port-version": 3,
   "description": "A lightweight and flexible C++ logging library",
   "homepage": "https://github.com/emilk/loguru",

--- a/ports/loguru/vcpkg.json
+++ b/ports/loguru/vcpkg.json
@@ -3,5 +3,13 @@
   "version-string": "2.1.0",
   "port-version": 2,
   "description": "A lightweight and flexible C++ logging library",
-  "homepage": "https://github.com/emilk/loguru"
+  "homepage": "https://github.com/emilk/loguru",
+  "features": {
+    "fmt": {
+      "description": "Build with fmt support in non-header-only mode",
+      "dependencies": [
+        "fmt"
+      ]
+    }
+  }
 }

--- a/ports/loguru/vcpkg.json
+++ b/ports/loguru/vcpkg.json
@@ -4,6 +4,16 @@
   "port-version": 2,
   "description": "A lightweight and flexible C++ logging library",
   "homepage": "https://github.com/emilk/loguru",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
   "features": {
     "fmt": {
       "description": "Build with fmt support in non-header-only mode",

--- a/ports/loguru/vcpkg.json
+++ b/ports/loguru/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "loguru",
   "version-string": "2.1.0",
-  "port-version": 2,
+  "port-version": 3,
   "description": "A lightweight and flexible C++ logging library",
   "homepage": "https://github.com/emilk/loguru",
   "dependencies": [

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4210,7 +4210,7 @@
     },
     "loguru": {
       "baseline": "2.1.0",
-      "port-version": 2
+      "port-version": 3
     },
     "lpeg": {
       "baseline": "1.0.2",

--- a/versions/l-/loguru.json
+++ b/versions/l-/loguru.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "78b95ff5e1c474dff22d728bba8920b3bbcfc0f4",
+      "version-string": "2.1.0",
+      "port-version": 3
+    },
+    {
       "git-tree": "c429f674d828c20881f410892d633b483e9d30dd",
       "version-string": "2.1.0",
       "port-version": 2

--- a/versions/l-/loguru.json
+++ b/versions/l-/loguru.json
@@ -1,7 +1,7 @@
 {
   "versions": [
     {
-      "git-tree": "78b95ff5e1c474dff22d728bba8920b3bbcfc0f4",
+      "git-tree": "ef57c8a1891b49ad8b5b8947f897a839cd108afe",
       "version-string": "2.1.0",
       "port-version": 3
     },

--- a/versions/l-/loguru.json
+++ b/versions/l-/loguru.json
@@ -1,8 +1,8 @@
 {
   "versions": [
     {
-      "git-tree": "ef57c8a1891b49ad8b5b8947f897a839cd108afe",
-      "version-string": "2.1.0",
+      "git-tree": "827f85e170a1d4548088fa36cf46817e146184d0",
+      "version": "2.1.0",
       "port-version": 3
     },
     {

